crates/wasmparser/src/validator/types.rs

```diff
@@ -816,6 +816,20 @@ impl ComponentEntityType {
             Self::Value(ty) => ty.type_size(),
         }
     }
+
+    /// The underlying `TypeId` for this entity type if one exists
+    pub fn id(&self) -> Option<TypeId> {
```

Comment:
I think we can remove this function if we stick to something from the above suggestion.